src/effects/nyquist/Nyquist.h

```diff
@@ -25,6 +25,11 @@ class wxTextCtrl;
 
 #define NYQUISTEFFECTS_VERSION wxT("1.0.0.0")
 
+// Protect Nyquist from selections greater than 2^31 samples (bug 439)
+#define NYQ_MAX_LEN (std::numeric_limits<long>::max())
```

Comment:
As I said 11 days ago, I don't agree with perpetuating [known bugs with known fixes](https://github.com/audacity/audacity/issues/1607) into new code.

Reply:
Steve, I will try to give that long neglected issue some attention.  But meantime, this proposed pull request addresses other problems, and is cutting and pasting the existing code in a way that does not make this issue worse.  Not better, but no worse.

So please consider this pull request on its own and be convinced that it is not causing any regressions.